core/drivers/pl022_spi.c

```diff
@@ -0,0 +1,505 @@
+			/* tx 1 packet */
+			write8(wdat[i++], pd->base + SSPDR);
+	}
+
+	gpio_set_value(pd->cs_gpio_pin, GPIO_LEVEL_HIGH);
+}
+
+static void pl022_tx16(struct spi_chip *chip, uint16_t *wdat,
+	size_t num_txpkts)
+{
+	size_t i = 0;
+	struct pl022_data *pd = container_of(chip, struct pl022_data, chip);
+
+	gpio_set_value(pd->cs_gpio_pin, GPIO_LEVEL_LOW);
+
+	while (i < num_txpkts) {
+		if (read8(pd->base + SSPSR) & SSPSR_TNF)
+			/* tx 1 packet */
+			write16(wdat[i++], pd->base + SSPDR);
+	}
+
+	gpio_set_value(pd->cs_gpio_pin, GPIO_LEVEL_HIGH);
+}
+
+static void pl022_rx8(struct spi_chip *chip, uint8_t *rdat,
+	size_t *num_rxpkts)
+{
+	size_t j = 0;
+	struct pl022_data *pd = container_of(chip, struct pl022_data, chip);
+
+	gpio_set_value(pd->cs_gpio_pin, GPIO_LEVEL_LOW);
+
+	do {
+		while ((read8(pd->base + SSPSR) & SSPSR_RNE)
+			&& (j < *num_rxpkts))
+			/* rx 1 packet */
+			rdat[j++] = read8(pd->base + SSPDR);
+	} while ((read8(pd->base + SSPSR) & SSPSR_BSY) && (j < *num_rxpkts));
+
+	*num_rxpkts = j;
+
+	gpio_set_value(pd->cs_gpio_pin, GPIO_LEVEL_HIGH);
+}
+
+static void pl022_rx16(struct spi_chip *chip, uint16_t *rdat,
+	size_t *num_rxpkts)
+{
+	size_t j = 0;
+	struct pl022_data *pd = container_of(chip, struct pl022_data, chip);
+
+	gpio_set_value(pd->cs_gpio_pin, GPIO_LEVEL_LOW);
+
+	do {
+		while ((read8(pd->base + SSPSR) & SSPSR_RNE)
+			&& (j < *num_rxpkts))
+			/* rx 1 packet */
+			rdat[j++] = read16(pd->base + SSPDR);
+	} while ((read8(pd->base + SSPSR) & SSPSR_BSY) && (j < *num_rxpkts));
+
+	*num_rxpkts = j;
+
+	gpio_set_value(pd->cs_gpio_pin, GPIO_LEVEL_HIGH);
+}
+
+static void pl022_print_peri_id(struct pl022_data *pd __maybe_unused)
+{
+	DMSG("Expected: 0x 22 10 #4 0");
+	DMSG("Read: 0x %x %x %x %x\n",
+		read32(pd->base + SSPPeriphID0),
+		read32(pd->base + SSPPeriphID1),
+		read32(pd->base + SSPPeriphID2),
+		read32(pd->base + SSPPeriphID3));
+}
+
+static void pl022_print_cell_id(struct pl022_data *pd __maybe_unused)
+{
+	DMSG("Expected: 0x 0D F0 05 B1");
+	DMSG("Read: 0x %x %x %x %x\n",
+		read32(pd->base + SSPPCellID0),
+		read32(pd->base + SSPPCellID1),
+		read32(pd->base + SSPPCellID2),
+		read32(pd->base + SSPPCellID3));
+}
+
+static void pl022_sanity_check(struct pl022_data *pd)
+{
+	assert(pd);
+	assert(pd->chip.ops);
+	assert(pd->base);
+	assert(pd->cs_gpio_base);
+	assert(pd->clk_hz);
+	assert(pd->speed_hz && pd->speed_hz <= pd->clk_hz/2);
+	assert(pd->mode <= SPI_MODE3);
+	assert(pd->data_size_bits == 8 || pd->data_size_bits == 16);
+
+	DMSG("SSPB2BTRANS: Expected: 0x2. Read: 0x%x\n",
+		read32(pd->base + SSPB2BTRANS));
+	pl022_print_peri_id(pd);
+	pl022_print_cell_id(pd);
+}
+
+static inline uint32_t pl022_calc_freq(struct pl022_data *pd,
+	uint8_t cpsdvr, uint8_t scr)
+{
+	return pd->clk_hz / (cpsdvr * (1 + scr));
+}
+
+static void pl022_calc_clk_divisors(struct pl022_data *pd,
+	uint8_t *cpsdvr, uint8_t *scr)
+{
+	unsigned int freq1 = 0, freq2 = 0;
+	uint8_t tmp_cpsdvr1, tmp_scr1, tmp_cpsdvr2 = 0, tmp_scr2 = 0;
```

Comment:
Changes made to only declare 1 variable per line, but please note that the variables are used outside the `for` loop as well.